### PR TITLE
Add guide on using retworkx for networkx users

### LINKED ISSUE #248: Make this library a drop in replacement for networkx
### What is the expected enhancement?

Right now I can't use this library since the class names are different from NetworkX and there's no reason they should be.
If the rest of the API matches, I can offer this as an extra in my package.

### PR DESCRIPTION
**Pull request by mtreinish**
This commit adds a new documentation guide for current networkx users on
how to start using retworkx or how to migrate existing networkx code. It
provides a high level explanation of the differences in behavior between
the 2 libraries and how you can you can switch between them.

Fixes #248

TODO:

- [x] Finish writing guide
- [x] Wait for #118 which adds the multigraph constructor option

### CODE REVIEW

**Comment by mtreinish** [comment]
I think there is still some more we can add here, but this is a decent starting point we can always expand the guide over time. But if people think there are obvious gaps we can add to this before merging.

File: docs/source/networkx.rst [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+   * - ``add_edge``
+     - :meth:`~retworkx.PyDiGraph.add_edge`
+     - retworkx requires 3 parameters be used the 2 node indices and the payload
+       (networkx works with either 2 or 3)
+   * - ``add_edges_from``
+     - :meth:`~retworkx.PyDiGraph.add_edges_from`,
+       :meth:`~retworkx.PyDiGraph.add_edges_from_no_data`,
+       :meth:`~retworkx.PyDiGraph.extend_from_edge_list`,
+       :meth:`~retworkx.PyDiGraph.extend_from_weighted_edge_list`
+     - retworkx requires a list of either a 3 or 2 tuple (depend on whether
```
Comment (kdk) [inline docs/source/networkx.rst:280]:
```suggestion
     - retworkx requires a list of either a 3 or 2 tuple (depending on whether
```

File: docs/source/networkx.rst [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+       ``extend_from*`` will create new nodes with a weight/data payload of
+       ``None`` if any node indices are missing.
+
+(note the retworkx version links to the :class:`~retworkx.PyDiGraph` version,
+but there are also equivalent :class:`~retworkx.PyGraph` methods available)
+
+Functionality Gaps
+==================
+
+networkx is a mature library that has wide user base and extensive feature set,
```
Comment (kdk) [inline docs/source/networkx.rst:292]:
```suggestion
networkx is a mature library that has a wide user base and extensive feature set,
```

**Review by kdk — APPROVED** [review]
Looks good on the whole. Few small comments.

File: docs/source/networkx.rst [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+retworkx (as the name implies) was inspired by networkx and the goal of the
+project is to provide a similar level of functionality and utility to what
+networkx offers but with much faster performance. However, because of
+limitations in the boundary between rust and python, different design
+decisions, and other differences the libraries are not identical.
+
+The biggest difference to keep in mind is networkx is a very dynamic in how it
+can be used. It allows you to treat a graph object associatively (like a python
+dictionary) and interact with the graph using the objects you're putting
```
[the remark this answers is not in the document]
Comment (mtreinish) [reply docs/source/networkx.rst:19]: Yeah, I had a hard time wording it, a networkx graph where a node is an index on a dict for itself. Do you have any thoughts on a clearer wording here? Or is it fine as is in combination with the rest of the paragraph and the code examples?

[the remark this answers is not in the document]
Comment (kdk) [reply docs/source/networkx.rst:19]: Maybe to move the examples up, something like:

```
The biggest difference to keep in mind is that networkx is very dynamic in how it
can be used, whereas retworkx, being written in rust, puts more constraints on how
you interact with graph object. networkx allows you to treat a graph object
like a python dictionary and interact with the graph using the objects you've inserted,
while with retworkx each node and edge is assigned an integer index
which must be used to access nodes and edges on the graph.

For example...
```

[the remark this answers is not in the document]
Comment (mtreinish) [reply docs/source/networkx.rst:19]: Ok, I gave that a try in: https://github.com/Qiskit/retworkx/pull/252/commits/93603104fb2ebb575f50459647c336ed949fcb93

File: docs/source/networkx.rst [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+       ``None`` if any node indices are missing.
+
+(note the retworkx version links to the :class:`~retworkx.PyDiGraph` version,
+but there are also equivalent :class:`~retworkx.PyGraph` methods available)
+
+Functionality Gaps
+==================
+
+networkx is a mature library that has a wide user base and extensive feature set,
+while retworkx by comparison is a much younger library and is missing a lot
```
Comment (elfrocampeador) [inline docs/source/networkx.rst:292]:
```suggestion
while retworkx, by comparison, is a much younger library and is missing a lot
```

**Review by elfrocampeador — CHANGES_REQUESTED** [review]
Pedantic copy editing, as is tradition

**Pull request merged**